### ISSUE TITLE
[BUG] The ability to ignore model field in request and just use current loaded model. (Ignore admin key checking if inline_model_loading set to false)

### ISSUE THREAD

**Issue by Originalimoc** [theroyallab/tabbyAPI#236]
### Problem

Currently there will be an error if requested model is not the loaded model and admin API is not provided.

### Solution

Add an option to ignore the model parameter and just use current loaded model instead.

### Explanation

NA

### Acknowledgements

- [X] I have looked for similar requests before submitting this one.
- [X] I understand that the developers have lives and my issue will be answered when possible.
- [X] I understand the developers of this program are human, and I will make my requests politely.

**Comment by Ph0rk0z** [comment]
This might be helpful for stuff that traditionally uses openAI api and requests something like 4o.

**Comment by Originalimoc** [comment]
Actually right now it can be acheived by setting "disable_auth: false" && "inline_model_loading: false".
But ideal situation is when "inline_model_loading == false", ignore admin key presence checking. A simple code reording would do.

**Comment by DocShotgun** [comment]
What exactly happens when you send an empty or non-listed model name with the request?

The behavior you are requesting has always been the tabby default. If it’s not working now, I suspect it’s a problem introduced by the addition of in-line model loading (which is disabled on my local setup).

**Comment by Originalimoc** [comment]
The auth check currently is before inline_model_loading check.

This code can be improved just by placing if check inline_model_loading first:
 
if inline_model_loading:
    if check_auth == "admin":
        load_new_model
else:
    ignore model field

**Comment by DocShotgun** [comment]
Does it throw an error if you request with a model name that doesn’t exist?

**Comment by Originalimoc** [comment]
Hmmm that's a separate issue? I never tried requesting a non-existing model.

**Comment by DocShotgun** [comment]
Ah ok, I misread when I was looking at this on my phone from work lol.

On re-reading, I'm not sure that what you are describing is necessarily a bug. It would be a design decision whether to just error and tell the requestor that they are trying to do something that is unauthorized (changing model without admin key) versus silently ignoring that part of the request and genning anyways.

**Comment by Originalimoc** [comment]
What would client expect if inline_model_loading is set to false anyway? Should that be a Bad Request error too? Currently it is not. The behavior is not consistent so I consider it an undesired feature. Add a option to select return HTTP bad request or just ignore model field if (model does not match && inline_model_loading == false) is best way IMO, of course when AUTH enabled just 401, no issue there. Currently it is also a silently ignoring.
Updated code would look like this:
```
async def load_inline_model(model_name: str, request: Request):
    """Load a model from the data.model parameter"""

    # Check if inline model loading is enabled
    if not config.model.inline_model_loading:
        if config.model.return_bad_request_on_missing_model:
            error_message = f"Inline model loading is disabled, cannot load model '{model_name}'."
            raise HTTPException(400, error_message)
        else:
            logger.info(
                f"Ignoring model field '{model_name}' because "
                '"inline_model_loading" is disabled in config.yml.'
            )
            return

    # Check if the user has admin permissions
    if not get_key_permission(request) == "admin":
        error_message = handle_request_error(
            f"Unable to switch model to {model_name} because "
            + "an admin key isn't provided",
            exc_info=False,
        ).error.message
        raise HTTPException(401, error_message)

    # Return if the model container already exists and the model is fully loaded
    if (
        model.container
        and model.container.model_dir.name == model_name
        and model.container.model_loaded
    ):
        return

    # Validate the model path
    model_path = pathlib.Path(config.model.model_dir) / model_name
    if not model_path.exists():
        if config.model.return_bad_request_on_missing_model:
            error_message = f"Model path '{str(model_path)}' not found."
            raise HTTPException(400, error_message)
        else:
            logger.warning(
                f"Could not find model path {str(model_path)}. Skipping inline model load."
            )
            return

    # Load the model and add draft directory
    await model.load_model(
        model_path,
        draft_model=config.draft_model.model_dump(include={"draft_model_dir"}),
    )
```
and config.yml will add this key:
```
return_bad_request_on_missing_model: true
```

**Comment by DocShotgun** [comment]
>What would client expect if inline_model_loading is set to false anyway? Should that be a Bad Request error too? Currently it is not. The behavior is not consistent so I consider it an undesired feature.

The only problem I see with the current way the code is organized, is that the check for whether in-line model loading is enabled happens after the admin check. So you will error out first with 401 from the admin check even if in-line model loading is disabled. Perhaps that check should just be moved down.

> Add a option to select return HTTP bad request or just ignore model field if (model does not match && inline_model_loading == false)

I'm hesitant to introduce yet another config arg, not sure what @bdashore3's thoughts are on this.

**Comment by Originalimoc** [comment]
I prefer return_bad_request_on_missing_model default to false = same behavior as now.

**Comment by bdashore3** [comment]
Solved in f9fffd42e0b8de61f775a12db0a5dfa9831b27a1 and bd9e78e19e19550d91b73881bf85b6e29249f7bb

The inline model behavior should have the same flow as traditional loading. This is basically a simple permission model where API users can't load new models and admins can.

The main concern that's present here is client messaging when inline loading can't be done. Since the client can't see the console, the API (erroring) is the only way to let the client know if something goes wrong. Server is currently running model A. User generates with model B in the model flag. The model doesn't load and simply warns the console. This leads the user to believe that the response was from model B instead of model A which causes a lot of confusion, especially in the model testing community.

The only exception which was mentioned here is if a user is providing a dummy model. These are used in clients that require certain models to be passed (such as gpt-4o) when generating a completion. That's where the new parameter `dummy_model_names` comes into play. Provide the name of those dummy models as exceptions and therefore any model client should work. The config instructions have been updated for these changes.

Following this approach allows the client and server to be on the same page when using an API key. If you want to disable this behavior entirely, disabling the auth system is an option so you don't have to worry about permissions.

Closing this issue as it's solved in a way that best fits the auth model.
